2048/2048/src/main/java/com/appcoins/eskills2048/model/RoomApiMapper.java

```diff
@@ -0,0 +1,44 @@
+package com.appcoins.eskills2048.model;
+
+import androidx.annotation.NonNull;
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import java.util.Objects;
+import org.json.JSONObject;
+import retrofit2.HttpException;
+
+
+
+public class RoomApiMapper {
+  private class Response{
+    Detail detail;
+  }
+  private class Detail{
+    String code;
+    String message;
+  }
+
+  public RoomResponseErrorCode mapException(@NonNull Throwable throwable){
+    RoomResponseErrorCode response;
+    if(throwable instanceof HttpException) {
+      HttpException exception = (HttpException) throwable;
```

Comment:
this is more a fun fact, but in kotlin, if you check a variable type (like you did on line 23) then it's smart cast to the checked type inside the if-clause (meaning you don't need to do line 24)

Reply:
kotlin ftw

Reply:
kotlin ftw

Reply:
kotlin ftw